### PR TITLE
adding functions to extract headers from context

### PR DESCRIPTION
**Pull request by ana-maria-ricardo**
Externalising functions that might be reused in other repos as well.  from:  ([link](https://github.com/ricardo-ch/orders-seda-producer/pull/18/files/704d9eb7fb9833dd6f4afcafe70f4ac8a2bf5ce9#discussion_r332091879))

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -40,3 +41,23 @@ func DefaultTracing(ctx context.Context, msg *sarama.ConsumerMessage) (opentraci
...
+
+	recordHeaders := make([]sarama.RecordHeader, 0, len(carrier))
+	for headerKey, headerValue := range carrier {
+		recordHeaders = append(recordHeaders, sarama.RecordHeader{Key: []byte(headerKey), Value: []byte(headerValue)})
+	}
+	return recordHeaders
+}
+
+// SerializeKafkaHeadersFromContext fetch some metadata from context and serialize it into a json map[string]string
+func SerializeKafkaHeadersFromContext(ctx context.Context) (string, error) {
```
Comment (FrancoisPoinsot) [inline options.go]: you probably need to Deserialize operation too

Comment (ana-maria-ricardo) [reply options.go]: added reverse functions as well

File: options.go [hunk 2]
```diff
@@ -40,3 +41,50 @@ func DefaultTracing(ctx context.Context, msg *sarama.ConsumerMessage) (opentraci
...
+	kafkaHeaders := tracing.InjectIntoCarrier(ctx)
+	kafkaHeadersJSON, err := json.Marshal(kafkaHeaders)
+
+	return string(kafkaHeadersJSON), err
+}
+
+// DeserializeContextFromKafkaHeaders fetch some metadata from context and serialize it into a json map[string]string
```
Comment (FrancoisPoinsot) [inline options.go]:
```suggestion
// DeserializeContextFromKafkaHeaders fetch tracing headers from json encoded carrier and returns the context
```

File: options.go [hunk 3]
```diff
@@ -40,3 +41,50 @@ func DefaultTracing(ctx context.Context, msg *sarama.ConsumerMessage) (opentraci
...
+	carrier := tracing.InjectIntoCarrier(ctx)
+
+	recordHeaders := make([]sarama.RecordHeader, 0, len(carrier))
+	for headerKey, headerValue := range carrier {
+		recordHeaders = append(recordHeaders, sarama.RecordHeader{Key: []byte(headerKey), Value: []byte(headerValue)})
+	}
+	return recordHeaders
+}
+
+// GetContextFromKafkaMessage  fetches opentracing headers from the kafka message
```
Comment (FrancoisPoinsot) [inline options.go]:
```suggestion
// GetContextFromKafkaMessage fetches opentracing headers from the kafka message
```

File: options.go [hunk 4]
```diff
@@ -40,3 +41,50 @@ func DefaultTracing(ctx context.Context, msg *sarama.ConsumerMessage) (opentraci
 		&map[string]interface{}{"offset": msg.Offset, "partition": msg.Partition, "key": string(msg.Key)},
 	)
 }
+
+// GetKafkaHeadersFromContext fetch some metadata from context and returns them in format []RecordHeader
```
Comment (FrancoisPoinsot) [inline options.go]:
```suggestion
// GetKafkaHeadersFromContext fetch tracing metadata from context and returns them in format []RecordHeader
```

Comment (ana-maria-ricardo) [reply options.go]: rephrasing done

File: options.go (hunk 3 above)
Comment (ana-maria-ricardo) [reply options.go]: rephrasing done

File: options.go (hunk 2 above)
Comment (ana-maria-ricardo) [reply options.go]: rephrasing done

**Pull request merged**